arduino-ide-extension/src/browser/contributions/init-libs-platforms.ts

```diff
@@ -0,0 +1,55 @@
+import { LocalStorageService } from '@theia/core/lib/browser';
+import { inject, injectable } from '@theia/core/shared/inversify';
+import { BoardsService, LibraryService } from '../../common/protocol';
+import { Contribution } from './contribution';
+
+@injectable()
+export class InitLibsPlatforms extends Contribution {
+  @inject(LocalStorageService)
+  private readonly localStorageService: LocalStorageService;
+  @inject(BoardsService)
+  private readonly boardsService: BoardsService;
+  @inject(LibraryService)
+  private readonly libraryService: LibraryService;
+
+  override async onReady(): Promise<void> {
+    const isFirstStartup = !(await this.localStorageService.getData(
+      InitLibsPlatforms.INIT_LIBS_AND_PACKAGES
+    ));
+    if (isFirstStartup) {
+      await this.localStorageService.setData(
+        InitLibsPlatforms.INIT_LIBS_AND_PACKAGES,
+        true
+      );
```

Comment:
Since you changed the logic with the `try{} catch{}`; why do not IDE2 sets this flag after the installation.

Potential issue:
 - Is it the first start? yes,
 - Store that the libs and AVR were installed,
 - Install the libs and AVR,
 - It fails,
 - Next IDE2 start; is it the first start? No, but nothing is installed.

What do you think?

Reply:
Related: https://github.com/arduino/arduino-ide/issues/784

Reply:
> This behavior is expected in my opinion.

Another use case:
 - User has a fresh env and will open the IDE2 for the very first time,
 - No internet connection,
 - Install will fail but IDE2 stores that the install was already run,
 - User starts the IDE2 the second time with an Internet connection,
 - Nothing will happen.
 
If this is expected, please proceed with the merge.

Reply:
Ah, that makes sense! But in that case, moving the initialization of INIT_LIBS_AND_PACKAGES is not enough. I will work on a fix! Thanks!